### PR TITLE
Fix template injection

### PR DESCRIPTION
**Pull request by stefanlogue**
This PR addresses some issues raised in #40 post-merge

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -15,12 +15,30 @@ const (
...
-		return config.DefaultPrefixes, nil, nil, true, defaultCommitTitleCharLimit, defaultMessageTemplate, defaultMessageWithTicketTemplate, nil
+		return LoadConfigReturn{
+			defaultMessageTemplate,
+			defaultMessageWithTicketTemplate,
+			config.DefaultPrefixes,
+			nil,
+			nil,
+			defaultCommitTitleCharLimit,
+			true,
+		}, nil
```
Comment (ccoVeille) [inline config.go:41]: It works, but it has many caveats.
Among them, I would like to mention:
- readability: what does the second nil state for?
- maintenance: you cannot easily add a field or reorder them
- source of bug: imagine you set the prefix in authors field
- you remove the logic of default value provided by the typed field in the struct

I would recommend you this


```suggestion
		return LoadConfigReturn{
			MessageTemplate: defaultMessageTemplate,
			MessageWithTicketTemplate: defaultMessageWithTicketTemplate,
			Prefixes: config.DefaultPrefixes,
			CommitTitleCharLimit: defaultCommitTitleCharLimit,
			ShowIntro: true,
		}, nil
```

File: config.go [hunk 2]
```diff
@@ -29,7 +47,15 @@ func loadConfig(fs afero.Fs) ([]huh.Option[string], []huh.Option[string], []huh.
...
 	if err != nil {
-		return nil, nil, nil, true, defaultCommitTitleCharLimit, defaultMessageTemplate, defaultMessageWithTicketTemplate, fmt.Errorf("error parsing config file: %w", err)
+		return LoadConfigReturn{
+			defaultMessageTemplate,
+			defaultMessageWithTicketTemplate,
+			nil,
+			nil,
+			nil,
+			defaultCommitTitleCharLimit,
+			true,
```
Comment (ccoVeille) [inline config.go:57]:
```suggestion
		return LoadConfigReturn{
			MessageTemplate: defaultMessageTemplate,
			MessageWithTicketTemplate: defaultMessageWithTicketTemplate,
			CommitTitleCharLimit: defaultCommitTitleCharLimit,
			ShowIntro: true,
```

File: config.go [hunk 3]
```diff
@@ -65,5 +88,13 @@ func loadConfig(fs afero.Fs) ([]huh.Option[string], []huh.Option[string], []huh.
...
-	return c.Prefixes.Options(), c.Coauthors.Options(), c.Boards.Options(), *c.ShowIntro, *c.CommitTitleCharLimit, messageTemplate, messageWithTicketTemplate, nil
+	return LoadConfigReturn{
+		messageTemplate,
+		messageWithTicketTemplate,
+		c.Prefixes.Options(),
+		c.Coauthors.Options(),
+		c.Boards.Options(),
+		*c.CommitTitleCharLimit,
+		*c.ShowIntro,
+	}, nil
```
Comment (ccoVeille) [inline config.go:99]: Same here, please use the fields name

File: config.go (hunk 1 above)
Comment (stefanlogue) [reply config.go]: Ah that makes sense, I wasn't aware of the default value logic with typed fields!

**Review by ccoVeille — APPROVED** [review]

**Pull request merged**